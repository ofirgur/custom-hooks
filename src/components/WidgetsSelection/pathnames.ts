export const PATHNAMES = {
  home: '/',
  tarotMemoryGame: 'tarotmemorygame',
  tarotQuizGame: 'tarotquizgame',
};
