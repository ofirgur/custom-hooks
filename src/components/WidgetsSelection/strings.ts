export const STRINGS = {
  TAROT_MEMORY_GAME: 'Tarot Memory Game',
  TAROT_QUIZ_GAME: 'Tarot Quiz Game',
  TEST_AND_STYLE_GAME: 'Test and Style Game',
};
