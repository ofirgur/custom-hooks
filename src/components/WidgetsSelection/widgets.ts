import { PATHNAMES } from './pathnames';
import { STRINGS } from './strings';

export const WIDGETS = [
  {
    key: 'TarotMemoryGame',
    name: 'tarotmemorygame',
    path: PATHNAMES.tarotMemoryGame,
    title: STRINGS.TAROT_MEMORY_GAME,
    icon: '',
  },
  {
    key: 'TarotQuizGame',
    name: 'tarotquizgame',
    path: PATHNAMES.tarotQuizGame,
    title: STRINGS.TAROT_QUIZ_GAME,
    icon: '',
  },
];
