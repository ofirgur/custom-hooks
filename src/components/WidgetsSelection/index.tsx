import React, { Suspense } from 'react';
import { BrowserRouter, Outlet, Route, Routes } from 'react-router-dom';
import WidgetsDashbord from '../WidgetsDashbord';
import { PATHNAMES } from './pathnames';
import { WIDGETS } from './widgets';

const TarotMemoryGame = React.lazy(() => import(/* webpackChunkName: "tarot-memory-game" */'../../widgets/TarotMemoryGame'));
const TarotQuizGame = React.lazy(() => import(/* webpackChunkName: "tarot-quiz-game" */'../../widgets/TarotQuizGame'));
const TestAndStyleGame = React.lazy(() => import(/* webpackChunkName: "test-and-style-game" */'../../widgets/TestAndStyleGame'));

const games = {
  [PATHNAMES.tarotMemoryGame]: TarotMemoryGame,
  [PATHNAMES.tarotQuizGame]: TarotQuizGame,
};

const WidgetsSelection = ({ widgetName }: { widgetName: string }) => {
  if (games[widgetName]) {
    const Widget = games[widgetName];
    return (
      <Suspense fallback={<>...</>}>
        <Widget />
      </Suspense>
    );
  }
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Outlet />}>
          <Route index element={<WidgetsDashbord widgets={WIDGETS} />} />
          <Route
            path={PATHNAMES.tarotMemoryGame}
            element={(
              <Suspense fallback={<>...</>}>
                <TarotMemoryGame />
              </Suspense>
            )}
          />
          <Route
            path={PATHNAMES.tarotQuizGame}
            element={(
              <Suspense fallback={<>...</>}>
                <TarotQuizGame />
              </Suspense>
            )}
          />
          <Route
            path={PATHNAMES.testandstylegame}
            element={(
              <Suspense fallback={<>...</>}>
                <TestAndStyleGame />
              </Suspense>
            )}
          />
        </Route>
      </Routes>
    </BrowserRouter>
  );
};

export default WidgetsSelection;
