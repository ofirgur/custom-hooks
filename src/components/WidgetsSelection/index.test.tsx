import * as React from 'react';
import { render } from '@testing-library/react';

import WidgetsSelection from './';

describe('WidgetsSelection', () => {
  it('should render correctly', () => {
    const { container } = render(<WidgetsSelection widgetName="tarotMemoryGame" />);
    expect(container).toMatchSnapshot();
  });
});
