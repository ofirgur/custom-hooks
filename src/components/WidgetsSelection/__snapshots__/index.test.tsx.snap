// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetsSelection should render correctly 1`] = `
<div>
  <div
    class="css-8gwbof"
  >
    <h2
      class="css-7c3lnj"
    >
      EMAIL ACTIVITY WIDGETS
    </h2>
    <div
      class="css-rm55dl"
    >
      <div
        class="css-1flu1tv"
      >
        Tarot Memory Game
      </div>
      <div
        class="css-1flu1tv"
      >
        Tarot Quiz Game
      </div>
      <div
        class="css-1flu1tv"
      >
        Test and Style Game
      </div>
    </div>
  </div>
</div>
`;
